RecoParticleFlow/PFClusterProducer/plugins/PFRecHitProducer.cc

```diff
@@ -100,9 +100,10 @@ void PFRecHitProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup
 }
 
 void PFRecHitProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
-  //The following says we do not know what parameters are allowed so do no validation
-  // Please change this to state exactly what you do use, even if it is no parameters
   edm::ParameterSetDescription desc;
-  desc.setUnknown();
-  descriptions.addDefault(desc);
+  edm::ParameterSetDescription psd;
+  psd.setUnknown();
```

Comment:
Using setUknown really defeats the purpose of the fillDescriptions.

Reply:
The issue I had in this particular class was that there were so many variations of these parameter set, that failed to find a super set that cover all (so end up declaring it as unknown). Same with PFClusterProducer

Reply:
The unknown parameter set descriptions are removed in 2da72f03d53a1bd5cb9fc6e0f72f9205dce0e900

Reply:
@stahlleiton just a word of caution about the (many) `addOptional` added in https://github.com/cms-sw/cmssw/commit/2da72f03d53a1bd5cb9fc6e0f72f9205dce0e900. Currently the `ConfDB` parsing  skips any `cms.optional` in the resulting cfi files, so if at some point it any of these need to be configured at HLT it won't be possible (unless a default is provided in the C++ source code).

Reply:
I made them optional because there were parameter sets with different content used in different parts of CMSSW. So I only used add for those parameters that were common among all parameter sets.

Reply:
Removed the optional condition from all the parameters I added in the fillDescription in c85415bc65ac6663b9a4d267d545c6402d82a540